src/backend/gpopt/CGPOptimizer.cpp

```diff
@@ -245,14 +257,34 @@ InitGPOPT()
 {
 	GPOS_TRY
 	{
-		return CGPOptimizer::InitGPOPT();
+		CGPOptimizer::InitGPOPT();
 	}
 	GPOS_CATCH_EX(ex)
 	{
 		if (GPOS_MATCH_EX(ex, gpdxl::ExmaGPDB, gpdxl::ExmiGPDBError))
 		{
 			PG_RE_THROW();
 		}
+
+		GPOS_TRY
+		{
+			CGPOptimizer::TerminateGPOPT(true);
+		}
+		GPOS_CATCH_EX(exc)
+		{
+			errstart(WARNING, exc.Filename(), exc.Line(), NULL, TEXTDOMAIN);
+			errfinish(errcode(ERRCODE_INTERNAL_ERROR), errprintstack(true));
+		}
+		GPOS_CATCH_END;
+
+		errstart(WARNING, ex.Filename(), ex.Line(), NULL, TEXTDOMAIN);
+		errfinish(errcode(ERRCODE_INTERNAL_ERROR), errprintstack(true));
+
+		if (OptimizerMemoryContext != NULL)
```

Comment:
This block is never executed, because errfinish() terminates the process.

Reply:
> errfinish() terminates the process

Why?! errstart(WARNING...

Reply:
> errfinish() terminates the process

No! Process is terminated by debugger when it is attaching to new process at https://github.com/arenadata/gpdb/blob/20537e17098e824f6296366478821ada8e98b67e/src/backend/utils/error/elog.c#L3686 on printing stack trace